hugegraph-core/src/main/java/com/baidu/hugegraph/schema/PropertyKey.java

```diff
@@ -196,12 +196,12 @@ public String clazz() {
                 valid = this.checkDataType(value);
                 break;
             case SET:
-                valid = value instanceof Set;
-                valid = valid && this.checkDataType((Set<?>) value);
+                valid = value instanceof Set && this.checkDataType((Set<?>) value);
+                valid = valid || this.checkDataType(value);
```

Comment:
prefer to use contains condition: `has('property-name', ConditionP.contains(value))`